django_forms_encoder/encoder.py

```diff
@@ -20,41 +20,46 @@ class DjangoFormsEncoder(DjangoJSONEncoder):
     def default(self, obj):
         """Encode django form."""
         if isinstance(obj, Form):
-            form: Dict[str, Any] = {
-                'fields': [],
-            }
-            if obj.non_field_errors():
-                form['form_errors'] = obj.non_field_errors()
-
-            for name, field in obj.fields.items():
-                field_obj = {
-                    'name': name,
-                    'label': field.label or name,
-                    'required': field.required,
-                }
-
-                if field.help_text:
-                    field_obj['help_text'] = field.help_text
-
-                if obj.errors.get(name):
-                    field_obj['errors'] = obj.errors.get(name)
-
-                if obj.is_bound and obj.cleaned_data.get(name) is not None:
-                    field_obj['value'] = obj.cleaned_data.get(name)
-
-                if isinstance(field.widget, widgets.Input):
-                    field_obj['widget'] = {
-                        'name': WidgetType.INPUT,
-                        'type': field.widget.input_type,
-                    }
-                elif isinstance(field.widget, widgets.Textarea):
-                    field_obj['widget'] = {
-                        'name': WidgetType.TEXTAREA,
-                    }
-                else:
-                    raise TypeError('Widget of type {} is not JSON serializable'.format(type(field.widget)))
-
-                form['fields'].append(field_obj)
-            return form
+            return form_to_dict(obj)
 
         return super().default(obj)
+
+
+def form_to_dict(obj):
+    """Convert django form to simple dictionary."""
```

Comment:
Most of the tests for `DjangoFormsEncoder` should be transformed to tests for `form_to_dict`.

Reply:
Fixed.